src/azul/service/responseobjects/hca_response_v5.py

```diff
@@ -482,6 +500,10 @@ def return_response(self):
         else:
             object_key = self._push_content()
             file_name = 'hca-manifest-' + object_key.rsplit('/', )[-1]
+
+        if self.format == 'full':
+            file_name = self.storage_service.get(object_key + '.name').decode('utf-8')
```

Comment:
This should gracefully handle the absence of a `.name` file.

Reply:
If you decode UTF8 you should specify that at encoding time as well. But 'utf-8' is the default so

```suggestion
            file_name = self.storage_service.get(object_key + '.name').decode()
```